### PR TITLE
add decrecaption warning when dpdata throws errors while parsing cp2k

### PR DESCRIPTION
**Pull request by robinzyb**

### CODE REVIEW

**Comment by njzjz** [comment]
#556 didn't throw any errors though

**Comment by robinzyb** [comment]
> #556 didn't throw any errors though

I don't know how he wrote the code for dpdata, however, I got errors when I use his files.
```python
import dpdata
dpdata.LabeledSystem(file_name="./55-20rfj2", fmt="cp2k/aimd_output")
```

```shell
{
	"name": "PendingDeprecationWarning",
	"message": "
Hi, you got an error from dpdata, 
please check if your cp2k files include full information,
otherwise its version is not supported by dpdata.
Try use dpdata plugin from cp2kdata package,
for details, please refer to
https://robinzyb.github.io/cp2kdata/
",
	"stack": "---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
File ~/anaconda3/envs/ectoolkit/lib/python3.8/site-packages/dpdata/plugins/cp2k.py:22, in CP2KAIMDOutputFormat.from_labeled_system(self, file_name, restart, **kwargs)
     21 xyz_file = sorted(glob.glob(f\"{file_name}/*pos*.xyz\"))[0]
---> 22 log_file = sorted(glob.glob(f\"{file_name}/*.log\"))[0]
     23 print(tuple(Cp2kSystems(log_file, xyz_file, restart)))

IndexError: list index out of range

During handling of the above exception, another exception occurred:

PendingDeprecationWarning                 Traceback (most recent call last)
/Users/robinzhuang/workdir/tmp/cp2kdata_testspace/Untitled-1.ipynb Cell 2 line 2
      <a href='vscode-notebook-cell:/Users/robinzhuang/workdir/tmp/cp2kdata_testspace/Untitled-1.ipynb#W1sZmlsZQ%3D%3D?line=0'>1</a> import dpdata
----> <a href='vscode-notebook-cell:/Users/robinzhuang/workdir/tmp/cp2kdata_testspace/Untitled-1.ipynb#W1sZmlsZQ%3D%3D?line=1'>2</a> dpdata.LabeledSystem(file_name=\"./55-20rfj2\", fmt=\"cp2k/aimd_output\")

File ~/anaconda3/envs/ectoolkit/lib/python3.8/site-packages/dpdata/system.py:281, in System.__init__(self, file_name, fmt, type_map, begin, step, data, convergence_check, **kwargs)
    279 if file_name is None:
    280     return
--> 281 self.from_fmt(
    282     file_name,
    283     fmt,
    284     type_map=type_map,
    285     begin=begin,
    286     step=step,
    287     convergence_check=convergence_check,
    288     **kwargs,
    289 )
    291 if type_map is not None:
    292     self.apply_type_map(type_map)

File ~/anaconda3/envs/ectoolkit/lib/python3.8/site-packages/dpdata/system.py:318, in System.from_fmt(self, file_name, fmt, **kwargs)
    316 if fmt == \"auto\":
    317     fmt = os.path.basename(file_name).split(\".\")[-1].lower()
--> 318 return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)

File ~/anaconda3/envs/ectoolkit/lib/python3.8/site-packages/dpdata/system.py:1162, in LabeledSystem.from_fmt_obj(self, fmtobj, file_name, **kwargs)
   1161 def from_fmt_obj(self, fmtobj, file_name, **kwargs):
-> 1162     data = fmtobj.from_labeled_system(file_name, **kwargs)
   1163     if data:
   1164         if isinstance(data, (list, tuple)):

File ~/anaconda3/envs/ectoolkit/lib/python3.8/site-packages/dpdata/plugins/cp2k.py:27, in CP2KAIMDOutputFormat.from_labeled_system(self, file_name, restart, **kwargs)
     25     return tuple(Cp2kSystems(log_file, xyz_file, restart))
     26 except :
---> 27     raise PendingDeprecationWarning(string_warning)

PendingDeprecationWarning: 
Hi, you got an error from dpdata, 
please check if your cp2k files include full information,
otherwise its version is not supported by dpdata.
Try use dpdata plugin from cp2kdata package,
for details, please refer to
https://robinzyb.github.io/cp2kdata/
"
}
```

**Comment by njzjz** [comment]
Firstly, move `1.out` to `1.log` as `*.log` is hard-coded in the code. (There should be an argument to adjust it.)

```sh
mv 1.out 1.log
```

Then there are no errors anymore. Catching an exception will not work.

```py
>>> import dpdata
>>> dpdata.LabeledSystem(file_name="./", fmt="cp2k/aimd_output")
Data Summary
Labeled System
-------------------
Frame Numbers      : 0
Atom Numbers       : 0
Including Virials  : No
Element List       :
-------------------
```

**Comment by robinzyb** [comment]
The old code returns empty dict even if nothing is matched. Extra lines are added to raise StopIteration when all matches is failed.

File: dpdata/plugins/cp2k.py [hunk 1]
```diff
@@ -4,29 +4,48 @@
...
+                tmp_virial,
+            ) = dpdata.cp2k.output.get_frames(file_name)
+            if tmp_virial is not None:
+                data["virials"] = tmp_virial
+            return data
+        # TODO: in the future, we should add exact error type here
+        # TODO: when pattern match is failed
+        # TODO: For now just use RuntimeError as a placeholder.
+        except RuntimeError:
+            print(string_warning)
```
Comment (njzjz) [inline dpdata/plugins/cp2k.py:51]: This ignores the error and does not exit the program. Is it expected?

Comment (njzjz) [reply dpdata/plugins/cp2k.py:51]: There may be two codes acceptable:

```py
except RuntimeError from e:
    raise PendingDeprecationWarning(string_warning) from e
```

or 

```py
except RuntimeError from e:
    warnings.warn(string_warning)
    raise e
```

File: dpdata/plugins/cp2k.py [hunk 2]
```diff
@@ -4,29 +4,48 @@
...
 
 @Format.register("cp2k/aimd_output")
 class CP2KAIMDOutputFormat(Format):
     def from_labeled_system(self, file_name, restart=False, **kwargs):
-        xyz_file = sorted(glob.glob(f"{file_name}/*pos*.xyz"))[0]
-        log_file = sorted(glob.glob(f"{file_name}/*.log"))[0]
-        return tuple(Cp2kSystems(log_file, xyz_file, restart))
+        try:
+            xyz_file = sorted(glob.glob(f"{file_name}/*pos*.xyz"))[0]
+            log_file = sorted(glob.glob(f"{file_name}/*.log"))[0]
```
Comment (njzjz) [inline dpdata/plugins/cp2k.py:22]: These two lines do not need to be inside the `try...except` block.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpdata/plugins/cp2k.py [hunk 3]
```diff
@@ -4,29 +4,48 @@
 from dpdata.cp2k.output import Cp2kSystems
 from dpdata.format import Format
 
+string_warning = """
+Hi, you got an error from dpdata,
+please check if your cp2k files include full information,
+otherwise its version is not supported by dpdata.
+Try use dpdata plugin from cp2kdata package,
```
Comment (njzjz) [inline dpdata/plugins/cp2k.py:11]:
```suggestion
Try to use dpdata plugin from cp2kdata package,
```

**Review by njzjz — APPROVED** [review]
LGTM. Just a grammar issue.

**Comment by njzjz** [comment]
@robinzyb, we only use Squash merge.

**Comment by njzjz** [comment]
I squashed and recommitted this PR in 7fb45d88452886572edba7f48269c79fb84fb22e and force-pushed to the devel branch.

**Comment by robinzyb** [comment]
> @robinzyb, we only use Squash merge.

sorry. didn't know the procedure. How should I do next?

**Comment by njzjz** [comment]
> How should I do next?

Nothing, as I've manually resolved it. But next time, you'd better select "squash and merge".

image

**Pull request merged**